rali/rali/source/node_ssd_random_crop.cpp

```diff
@@ -0,0 +1,194 @@
+                                                                                                          _dest_height(_outputs[0]->info().height_batch())
+{
+    _crop_param = std::make_shared<RaliRandomCropParam>(_batch_size);
+    _is_ssd     = true;
+}
+
+void SSDRandomCropNode::create_node()
+{
+    _crop_width_val.resize(_batch_size);
+    _crop_height_val.resize(_batch_size);
+    _x1_val.resize(_batch_size);
+    _y1_val.resize(_batch_size);
+    _x2_val.resize(_batch_size);
+    _y2_val.resize(_batch_size);
+    _iou_range.resize(_batch_size);
+    if (_node)
+        return;
+
+    if (_dest_width == 0 || _dest_height == 0)
+        THROW("Uninitialized destination dimension")
+
+    _crop_param->create_array(_graph);
+    _node = vxExtrppNode_CropPD(_graph->get(), _inputs[0]->handle(), _src_roi_width, _src_roi_height, _outputs[0]->handle(), _crop_param->cropw_arr,
+                                _crop_param->croph_arr, _crop_param->x1_arr, _crop_param->y1_arr, _batch_size);
+
+    vx_status status;
+    if ((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
+        THROW("Error adding the crop resize node (vxExtrppNode_ResizeCropbatchPD    ) failed: " + TOSTR(status))
+}
+
+inline double ssd_BBoxIntersectionOverUnion(const BoundingBoxCord &box1, const BoundingBoxCord &box2, bool is_iou = false) 
+{
+    double iou;
+    float xA = std::max(box1.x, box2.x);
+    float yA = std::max(box1.y, box2.y);
+    float xB = std::min(box1.x + box1.w, box2.x + box2.w);
+    float yB = std::min(box1.y + box1.h, box2.y + box2.h);
+
+    float intersection_area = std::max((float)0.0, xB - xA) * std::max((float)0.0, yB - yA);
+
+    float box1_area = box1.h * box1.w;
+    float box2_area = box2.h * box2.w;
+
+    if(is_iou)
+        iou = intersection_area / float(box1_area + box2_area - intersection_area);
+    else
+        iou = intersection_area / float(box1_area);
+
+    return iou;
+}
+
+void SSDRandomCropNode::update_node()
+{
+    _crop_param->set_image_dimensions(_inputs[0]->info().get_roi_width_vec(), _inputs[0]->info().get_roi_height_vec());
+    _crop_param->update_array();
+    // std::cerr<<"\n batch_size:: "<<_batch_size<<"\n meta array size:: "<<_meta_data_info->size();
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(0, 6);
+    const std::vector<std::pair<float, float>> IOU = {std::make_pair(0.0f, 1.0f), std::make_pair(0.1f, 1.0f), std::make_pair(0.3f, 1.0f),
+                                            std::make_pair(0.5f, 1.0f), std::make_pair(0.45f, 1.0f), std::make_pair(0.35f, 1.0f), std::make_pair(0.0f, 1.0f) };   
+    int sample_option;
+    std::pair<float, float> iou;
+    float min_iou, max_iou;
+    in_width = _crop_param->in_width;
+    in_height = _crop_param->in_height;
+    bool invalid_bboxes = true;
+    _entire_iou = true;
+    BoundingBoxCord crop_box, jth_box;
+    int bb_count;
```

Comment:
Codacy found an issue: [The scope of the variable 'bb_count' can be reduced.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6302037)